python/ray/data/dataset.py

```diff
@@ -89,6 +90,9 @@
 # Whether we have warned about using slow Dataset transforms.
 _slow_warned = False
 
+# Whether the dataset usage is recorded or not.
+_usage_recorded = False
```

Comment:
Can we move this into the common util lib?

Reply:
Furthermore, we should guard and only report on the driver process for performance reasons.